pkg/kubelet/certificate/kubelet.go

```diff
@@ -61,7 +61,17 @@ func NewKubeletServerCertificateManager(kubeClient clientset.Interface, kubeCfg
 			StabilityLevel: compbasemetrics.ALPHA,
 		},
 	)
+	var certificateRenewFailure = compbasemetrics.NewGauge(
```

Comment:
I think a [Counter](https://godoc.org/github.com/kubernetes/component-base/metrics#Counter) would be a more appropriate metric than a Gauge. A gauge is like a snapshot of the current state of something. While I feel like renewal failures are events that we should keep a count of.

IMO this should be a counter with a label for failure type: "failed to load client cert", "failed to contact master", "returned cert was not signed", ect. 

Then below when you actually observe these failures you can Inc() with those labels. That way we can track error rates and what the errors are more easily.

Reply:
So this would mean changing all your .Set(...) calls below.

Reply:
Heh. I initially wrote this using a counter... I'll change this PR to use a counter and rebase on top of your PR when it merges.